code/datums/helper_datums/synthsprites.dm

```diff
@@ -0,0 +1,64 @@
+
+/* 
+
+Just some quick documentation about how this works
+
+Synthckey is the players ckey which is represented under the synths name in the json, this is required to have
+synthicon is just the sprites name in the dmi
+aichassisicon is the ai chassis icon sprite name
+aiholoicon is the ai hologram icon sprite name
+paiicon is the pai icon sprite name
+
+*/
+
+
+
+/datum/custom_synth
+	var/synthname = ""
+	var/synthckey = ""
+	var/synthicon = "robot"
+	var/aichassisicon = ""
+	var/aiholoicon = ""
+	var/paiicon = ""
+
+
+
+/proc/loadsynths_from_json()
+	var/list/customsynthsprites = list()
+	try
+		customsynthsprites = json_decode(return_file_text("config/customsynths.json"))
+	catch(var/exception/ej)
+		log_debug("Error: Warning: Could not load custom synth config, as customsynths.json is missing - [ej]")
+		return
+
+	robot_custom_icons = list()
+	customsynthsprites = customsynthsprites["synths"]
+	for (var/synthsprite in customsynthsprites)
+		var/datum/custom_synth/synth = new()
+		synth.synthckey = customsynthsprites[synthsprite]["ckey"]
```

Comment:
`synth.synthname` is not set in this loop. It should probably be `synth.synthname = synthsprite`